zanata-war/src/main/java/org/zanata/search/HTextFlowTargetIndexingStrategy.java

```diff
@@ -60,4 +68,62 @@ protected ScrollableResults queryResults(int ignoredOffset, FullTextSession sess
         query.setFetchSize(Integer.MIN_VALUE);
         return query.scroll(ScrollMode.FORWARD_ONLY);
     }
+
+    public void reindexForProject(HProject project, FullTextSession session,
+            AsyncTaskHandle handle) {
+        // it must use the same session in the DAO and to do the indexing
+        HTextFlowTargetStreamingDAO dao =
+                new HTextFlowTargetStreamingDAO(HTextFlowTarget.class, session);
+        ScrollableResults scrollableResults =
+                dao.getTargetsWithAllFieldsEagerlyFetchedForProject(
+                        project);
+        reindexScrollableResultSet(session, scrollableResults, handle);
+    }
+
+    private static void reindexScrollableResultSet(FullTextSession session,
```

Comment:
Just realize when I checked the code. This method is not async but the caller is: org.zanata.service.impl.IndexingServiceImpl#reindexHTextFlowTargetsForProject

I found a typo in the method name above. I will update it.